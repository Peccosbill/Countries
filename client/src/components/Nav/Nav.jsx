import React from "react";
import { NavLink } from "react-router-dom";
// import SearchBar from "./SearchBar";
import styled from "styled-components";
import styles from "./Nav.module.css";

const Navigator = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

function Nav() {
  return (
    <Navigator>
      <h1>Countries</h1>
      {/* <SearchBar /> */}
      <NavLink className={styles.link} to="/addactivity">Agregar actividad</NavLink>
    </Navigator>
  );
}

export default Nav;
