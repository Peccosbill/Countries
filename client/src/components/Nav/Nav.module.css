.nav {
  background: var(--independence);
  width: 100%;
  height: 4.3rem;
  box-shadow: 0 0 2rem var(--rich-black);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logoAndTitle {
  text-decoration: none;
  color: var(--white);
  height: 3.8rem;
  padding: 0 0.5rem;
  margin-left: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0.2rem;
  background: var(--prussian-blue);
  box-shadow: 0 0 1rem var(--rich-black);
}

.logoAndTitle h1 {
  margin-left: 0.5rem;
  color: var(--blue-munsell);
}

.logoFlag {
  width: 3.5rem;
}

.logo {
  width: 12rem;
}

.link {
  text-decoration: none;
  text-align: center;
  color: #fff;
  padding: 0.5rem;
  margin-right: 2rem;
  height: 2.2rem;
  background: var(--candy-pink);
  border-radius: 0.2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  transition: 0.3s ease-in-out;
}

.link:hover {
  /* background: var(--blue-munsell); */
  filter: brightness(80%);
}

.plus {
  width: 1.5rem;
  margin-left: 0.3rem;
}
