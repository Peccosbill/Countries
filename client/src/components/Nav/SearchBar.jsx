import React from "react";

function SearchBar() {
  return (
    <div>
      <form>
        <input type="text"></input>
        <button type="submit">Buscar</button>
      </form>
    </div>
  );
}

export default SearchBar;
