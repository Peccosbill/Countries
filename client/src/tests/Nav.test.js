import "@testing-library/jest-dom";
import { NavLink } from "react-router-dom";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Nav from "../components/Nav/Nav";

configure({ adapter: new Adapter() });

describe("Probamos el componente <Nav />", () => {
  // eslint-disable-next-line no-unused-vars
  let nav = shallow(<Nav />);

  beforeEach(() => {
    nav = shallow(<Nav />);
  });

  it("debería mostrar <Nav /> correctamente", () => {
    expect(nav).toMatchSnapshot();
  });

  it("Debería renderizar dos <NavLink to='' />", () => {
    expect(nav.find(NavLink).length).toBeGreaterThanOrEqual(2);
  });

  it("Debería tener un NavLink con el texto AÑADIR ACTIVIDAD que cambie a la ruta '/home'", () => {
    expect(nav.find(NavLink).at(0).prop("to")).toEqual("/home");
  });

  it("Debería tener un NavLink con el texto AÑADIR ACTIVIDAD que cambie a la ruta '/home/addactivity'", () => {
    expect(nav.find(NavLink).at(1).prop("to")).toEqual("/home/addactivity");
    expect(nav.find(NavLink).at(1).text()).toEqual("AÑADIR ACTIVIDAD");
  });
});
