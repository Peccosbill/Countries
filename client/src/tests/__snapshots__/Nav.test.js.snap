// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probamos el componente <Nav /> deber√≠a mostrar <Nav /> correctamente 1`] = `ShallowWrapper {}`;
